mdtk/degradations.py

```diff
@@ -111,14 +110,47 @@ def pitch_shift(excerpt, min_pitch=MIN_PITCH, max_pitch=MAX_PITCH,
         A copy of the given excerpt, with the pitch of one note changed.
     """
     if excerpt.note_df.shape[0] == 0:
-        warnings.warn('WARNING: No notes to remove. Returning None.',
+        warnings.warn('WARNING: No notes to pitch shift. Returning None.',
                       category=UserWarning)
         return None
     
```

Comment:
Been staring at the bit of code to get `valid_notes` and wondering why it is necessary.

If you use the same tactic as when `distribution` is `None`, i.e. `while degraded.note_df.loc[note_index, 'pitch'] == pitch:`, and you just clip the result (force a min and max), surely it doesn't matter which note you pick to try and pitch shift.

Reply:
If the user gives us a distribution, min and max change for each pitch. That's what max_to_sample and min_to_sample are: the min and max for a given pitch, taking into account min_pitch, max_pitch, and distribution. This is still fine. However, if the use gives us a 1-sided distribution (for example, if a user wants us to sample overtones of +1 or 2 octaves from the ground truth pitch, but no undertones), and the original note is close to that end of [min_pitch, max_pitch], it may lead to a note that cannot be shifted.

I expect this to be somewhat rare, but one-sided samples may be relatively common. There is currently no explicit test for this case, I've realized.

(Also, sorry--I commented this yesterday but apparently I did something weird and it didn't go through...)

Reply:
I understand, I think there's some argument this would work much quicker with a 'blockheaded' approach of simply sampling a pitch, shifting, checking, then repeating up to some maximum number of times...but your implementation is more precise!